pandas/core/internals/blocks.py

```diff
@@ -1934,13 +1934,39 @@ def to_native_types(
 class DatetimeLikeBlockMixin(HybridMixin, Block):
     """Mixin class for DatetimeBlock, DatetimeTZBlock, and TimedeltaBlock."""
 
-    @property
-    def _holder(self):
-        return DatetimeArray
+    _dtype: np.dtype
+    _holder: Type[Union[DatetimeArray, TimedeltaArray]]
 
-    @property
-    def fill_value(self):
-        return np.datetime64("NaT", "ns")
+    @classmethod
+    def _maybe_coerce_values(cls, values):
```

Comment:
this seems pretty generic. do we need the override methods of this? (at all?)

Reply:
This shares is used by DTBlock/TDBlock, which in master each implement their own.  We still have to override on DatetimeTZBlock, but wont if/when the others are backed by EAs